ui/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/DeploymentManager.scala

```diff
@@ -11,24 +11,33 @@ import scala.concurrent.Future
 
 trait DeploymentManager extends AutoCloseable {
 
-  //TODO: savepointPath is very flink specific, how can we handle that differently?
+  /**
+    * This method is invoked separately before deploy, to be able to give user quick feedback, as deploy (e.g. on Flink) may take long time
+    */
+  def validate(processVersion: ProcessVersion, deploymentData: DeploymentData, canonicalProcess: CanonicalProcess): Future[Unit]
```

Comment:
why do we return future here ? I mean we rather return cats Validated when when we do synchronous validation

Reply:
it's not synchronous (at least not necessarily)